enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+---
+title: single-node-developer-cluster-profile
+authors:
+  - "@rkukura"
+reviewers:
+  - "@deads2k"
+  - "@LalatenduMohanty"
+  - "@wking"
+approvers:
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+creation-date: 2020-09-18
+last-updated: 2020-10-05
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Single Node Developer Cluster Profile
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Add a new single-node-developer [cluster
+profile](https://github.com/openshift/enhancements/blob/master/enhancements/update/cluster-profiles.md)
+that defines the set of OpenShift components, and the configuration
+thereof, applicable to single-node OpenShift Container Platform
+clusters intended for application development as well as for learning
+and exploring OpenShift, but never for production deployment of
+applications. This cluster profile will be utilized in producing the
+Code Ready Containers (CRC) product that runs in a single VM on a
+developer’s workstation or laptop.
+
+In a cluster deployed using this profile, a single node serves as both
+the cluster’s control plane and as a worker node. The cluster profile
+will support the OpenShift components needed for application
+development, including the operator hub and console. In order to
+minimize system resource requirements, the cluster profile will
+exclude production-oriented features such as high availability of the
+control plane components, machine upgradability, and advanced
+telemetry and logging.
+
+This cluster profile will result in a new topology of OpenShift
+components to be officially supported for its intended usage in the
+CRC product. Once the profile is introduced, all teams developing
+OpenShift components will need to be aware of how their components are
+configured in this profile. For many components, this will be simply a
+matter of whether the component is included or excluded. For others,
+specialized configurations and/or features may be used in this
+profile, possibly requiring development and ongoing maintenance
+efforts.
+
+## Motivation
+
+The CRC product’s VM image is currently built using [Single Node
+Cluster (SNC)](https://github.com/code-ready/snc) tooling. The snc.sh
+shell script runs the OpenShift installer to deploy a running cluster
+into a libvirt/KVM environment. One VM serves as the bootstrap node,
```

Comment:
will the libvirt/KVM be only supported virtualization? how about windows and mac?

Reply:
SNC is only run on linux and uses openshift-install+libvirt. The image which was generated by SNC is then used by CRC, which runs on linux/windows/macos. For SNC, we'll only be using libvirt/KVM in the foreseeable future.

Reply:
Nothing about the single-node-developer cluster profile is specific to libvirt/KVM being the install target. As @cfergeau describes, this is just a detail of how SNC currently functions.